src/server/handlers.rs

```diff
@@ -0,0 +1,388 @@
+        // For empty payloads, just fetch the latest version
+        let req: GetModelRequest = if msg.payload.is_empty() {
+            GetModelRequest { version: None }
+        } else {
+            match serde_json::from_reader(std::io::Cursor::new(msg.payload)) {
+                Ok(r) => r,
+                Err(e) => {
+                    self.send_error(
+                        msg.reply,
+                        format!("Unable to parse get model request: {e:?}"),
+                    )
+                    .await;
+                    return;
+                }
+            }
+        };
+
+        let manifests: StoredManifest = match self.store.get(lattice_id, name).await {
+            Ok(Some(m)) => m,
+            Ok(None) => {
+                self.send_reply(
+                    msg.reply,
+                    // SAFETY: We construct this data internally so we are fine
+                    serde_json::to_vec(&GetModelResponse {
+                        result: GetResult::NotFound,
+                        message: format!("Model with the name {name} not found"),
+                        manifest: None,
+                    })
+                    .unwrap(),
+                )
+                .await;
+                return;
+            }
+            Err(e) => {
+                error!(error = %e, "Unable to store updated data");
+                self.send_error(msg.reply, "Internal storage error".to_string())
+                    .await;
+                return;
+            }
+        };
+        let reply = match req.version {
+            Some(version) => {
+                if let Some(current) = manifests.get_version(&version) {
+                    GetModelResponse {
+                        manifest: Some(current.to_owned()),
+                        result: GetResult::Success,
+                        message: format!("Fetched model {name} with version {version}"),
+                    }
+                } else {
+                    self.send_reply(
+                        msg.reply,
+                        // SAFETY: We construct this data internally so we are fine
+                        serde_json::to_vec(&GetModelResponse {
+                            result: GetResult::NotFound,
+                            message: format!("Model {name} with version {} doesn't exist", version),
+                            manifest: None,
+                        })
+                        .unwrap(),
+                    )
+                    .await;
+                    return;
+                }
+            }
+            None => GetModelResponse {
+                manifest: Some(manifests.get_current().to_owned()),
+                result: GetResult::Success,
+                message: format!("Fetched model {name}"),
+            },
+        };
+        // SAFETY: We _just_ deserialized this from the store above, so we should be just fine
+        self.send_reply(msg.reply, serde_json::to_vec(&reply).unwrap())
```

Comment:
In general, is there a real downside to `unwrap_or_default` in all these safety cases?

Reply:
Actually, there probably isn't and it would be more graceful. Let me go update those